doc/auto_storage.md

```diff
@@ -516,23 +537,53 @@ system (so the same conditions can be matched by a disk, a partition, an LVM dev
-where) Agama should calculate and create the extra partitions needed for booting.
+Using a `boot` entry makes it possible to configure whether (and where, using an alias) Agama
+should calculate and create the extra partitions needed for booting.
 
-If the proposal is not used, Agama will always try to calculate and create those partitions taking
-the location of the root file system as a reference. That's the same approach that AutoYaST has
-followed for years.
+## Keeping an Existing File System or Encryption Layer
 
-## Using the Automatic Proposal
+The entries for both `encryption` and `filesystem` contain a flag `reuse` with a default value of
+false. It can be used in combination with `search` to specify the device must not be re-encrypted
+or re-formatted.
+
+## Deleting and Shrinking Existing Devices
 
-Agama can rely on the process known as Guided Proposal to calculate all the needed partitions, LVM
-devices and file systems based on some general product settings and some user preferences. That
-mechanism can also be used as part of the profile and will be executed as a last step, after
-processing all the explicit sections that describe devices.
+The storage proposal must make possible to define what to do with existing partitions and logical
+volumes. Even with existing MD RAIDs or LVM volume groups.
 
-The `guided` section conforms to the following specification.
+In order to provide the same capabilities than the Guided proposal (see below) it must be possible
+to specify that a given partition (or LVM logical volume) must be:
+
+  - Deleted if needed to make space for the newly defined devices
+  - Deleted in all cases
+  - Shrunk to the necessary size to make space for new devices
+  - Shrunk to a given size, maybe a range, in all cases (not really possible in the current Guided
+    Proposal)
+
+It would be desirable to have the possibility of expressing some combinations of the above, like
+"try to shrink it to make space but proceed to delete it if shrinking it is not enough".
+
+For partitions that are marked to be deleted (either mandatory or on demand), it makes no sense to
+specify any other usage (like declaring a file system on it). But resizing a partition is more on
+the grey area. Often some partitions or logical volumes are resized only to make space for the
+declared devices. But since resizing is not a destructive operation it can also make sense to
+declare a given partition must be resized and then formatted.
+
+The exact syntax to specify all the possible actions is still under discussion. The current document
+shows how a `delete` attribute could be used in combination with `search` to specify the devices to
+delete. An alternative could be to have separate section for each device to specify how to delete
+and/or shrink its partitions or logical volumes, instead of integrating that information into the
+corresponding `partitions` or `logical_volumes` attributes.
```

Comment:
Regarding this, I think I would kill all the current `delete` flags. In fact, the current `spaceMaker` only deletes partitions (and maybe logical volumes in the future). Deleting MDs and VGs is not considered by the SpaceMaker or by AutoYaST nowadays.

Then I would add a "space_actions" (tentative name) to everything that could contain partitions (`Drive` and `MdRaid`) and maybe also in the future also to volume groups in order to decide what to do with the existing LVs.

Each entry of that section would consist on one `search` (same syntax and context than the corresponding `partitions` section), one (optional) `delete` and one (also optional) `shrink`.

Example - delete everything and create a root:

```json
"drives": [
   {
     "partitions": [
        { 
           "size": { "min": "1 GiB" },
           "filesystem": { "path": "/", "type": "btrfs" }
        }
     ],
     "space_actions": [
        { 
           "search": {},
           "delete": true
        }
      ]
   }
]
```

Example - try resizing partitions first and deleting them is that's not enough:

```json
"drives": [
   {
     "partitions": [
        { 
           "size": { "min": "1 GiB" },
           "filesystem": { "path": "/", "type": "btrfs" }
        }
     ],
     "space_actions": [
        { 
           "search": {},
           "delete": "onDemand",
           "shrink": "onDemand"
        }
      ]
   }
]
```


Example - Keep the first partition, resizing it if needed, and delete the rest

```json
"drives": [
   {
     "partitions": [
        { 
           "size": { "min": "1 GiB" },
           "filesystem": { "path": "/", "type": "btrfs" }
        }
     ],
     "space_actions": [
        { 
           "search": { "max": 1 },
           "shrink": "onDemand"
        },
        { 
           "search": {}
           "delete": true
        }
      ]
   }
]
```

Example - Keep the biggest partition, resizing it to a certain size, and delete the rest if needed

```json
"drives": [
   {
     "partitions": [
        { 
           "size": { "min": "1 GiB" },
           "filesystem": { "path": "/", "type": "btrfs" }
        }
     ],
     "space_actions": [
        { 
           "search": { "sort": { "property": "sizeKib", "order": "desc" }, "max": 1 },
           "shrink": "15 GiB"
        },
        { 
           "search": {}
           "delete": "onDemand"
        }
      ]
   }
]
```

Reply:
Re-reading my proposal... I don't really like it. But I think some points can be taken as valid. For example:

>  Shrinking a partition to make space would be the only kind of resizing I would support. No support for enlarging existing partitions or for mounting an existing partition and also changing its size.